### PR TITLE
feat: use CH deduping instead

### PR DESCRIPTION
**Pull request by paolodamico**
README explains it better.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { Plugin, PluginEvent } from '@posthog/plugin-scaffold'
 import { createHash, randomUUID } from 'crypto'
 import { URLSearchParams } from 'url'
 
+const NAMESPACE_OID = '6ba7b812-9dad-11d1-80b4-00c04fd430c8' // From RFC #4122
```
Comment (yakkomajuri) [inline index.ts]: is this your own internal thing? can we generalize this plugin a bit so that it is maybe usable by others?

File: index.ts [hunk 2]
```diff
@@ -14,64 +16,65 @@ const stringifyEvent = (event: PluginEvent): string => {
...
-        // Check if event is in cache (hashed to limit memory usage)
+        // Create a hash of the relevant properties of the event
         const stringifiedProps = config.dedupMode === 'All Properties' ? `_${JSON.stringify(event.properties)}` : ''
         const hash = createHash('sha1')
-        const eventKey = hash
-            .update(`${event.team_id}_${event.distinct_id}_${event.event}_${event.timestamp}${stringifiedProps}`)
-            .digest('hex')
+        const eventKeyBuffer = hash
+            .update(
+                `${NAMESPACE_OID}_${event.team_id}_${event.distinct_id}_${event.event}_${event.timestamp}${stringifiedProps}`
```
Comment (yakkomajuri) [inline index.ts:69]: eventually I'd love for this to be configurable so other users can set their desired values - but for now happy to get it in as is

**Review by yakkomajuri — APPROVED** [review]

File: index.ts (hunk 1 above)
Comment (paolodamico) [reply index.ts]: It is not, it's from the UUID5 RFC (https://datatracker.ietf.org/doc/html/rfc4122), I'll add the comment to clarify.

File: index.ts (hunk 2 above)
Comment (paolodamico) [reply index.ts:69]: there's the config options on whether to use only key properties, but happy to generalize more, in a separate PR, what are you thinking?

**Pull request merged**